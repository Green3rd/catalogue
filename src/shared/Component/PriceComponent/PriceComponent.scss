@import "../../common.scss";

.PriceComponent {
  color: $priceColor;
  font-size: 1.2em;

  &--bold {
    .PriceComponent__Price {
      font-weight: bold;
    }

    font-size: 1.3em;
  }

  &--grey {
    float: right;
    color: $greyTextColor;
    font-size: 1em;
  }

  &__PriceContainer--block {
    display: block;
  }

  &__Currency {
    font-size: 0.7em;
    color: $greyTextColor;
  }

  &__CrossOutContainer {
    position: relative;
    margin-right: 8px;
  }

  &__CrossOut {
    color: $greyTextColor;
    font-size: 0.65em;
    padding: 0 5px;
    position: relative;
    bottom: 3px;

    &:before {
      content: "";
      position: absolute;
      right: 0;
      top: 81%;
      width: 110%;
      border-top: 2px solid;
      border-color: #e12d2d;
      -webkit-transform: rotate(-20deg);
      transform: rotate(-20deg);

      @supports (-webkit-touch-callout: none) {
        // Ios only
        right: -1px;
        top: 51%;
      }
    }
  }

  &__Discount {
    background-color: $priceColor;
    border-radius: 4px;
    color: $mainTextColor;
    padding: 3px 8px;
    font-size: 0.55em;
    font-weight: bold;
    margin-left: 12px;
    bottom: 3px;
    position: relative;
    float: right;
  }
}