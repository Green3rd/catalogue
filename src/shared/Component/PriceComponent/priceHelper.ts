// Not support negative number.
export const numberWithCommas = (x: number) => {
  let result = new Intl.NumberFormat('en-us', { style: 'currency', currency: 'USD' }).format(x);
  if (result.slice(-2) === "00") result = result.substring(0, result.length - 3); // Remove '.00'
  return result.substring(1); // Remove a currency symbol;
};
