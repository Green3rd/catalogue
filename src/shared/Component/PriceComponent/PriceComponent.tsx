import React from "react";
import { generateClassName } from "shared/utils/reactClassName";
import { numberWithCommas } from "./priceHelper";
import "./PriceComponent.scss";

interface Props {
  price: Money;
  crossOutprice?: number;
  bold?: boolean;
  grey?: boolean;
  percentDiscount?: number;
}

const PriceComponent = (props: Props) => {
  const { price, bold, grey, crossOutprice, percentDiscount } = props;
  const crossOut = crossOutprice && (
    <span className="PriceComponent__CrossOutContainer">
      <span className="PriceComponent__CrossOut">
        {numberWithCommas(crossOutprice)}
      </span>
    </span>
  );

  const discount = percentDiscount && (
    <span className="PriceComponent__Discount">ลด {percentDiscount}%</span>
  );


  return (
    <span className={generateClassName("PriceComponent", [bold ? "bold" : "", grey ? "grey" : ""])}>
      {discount === 0 ? '' : discount}
      <span className={generateClassName("PriceComponent__PriceContainer", [percentDiscount ? "block" : ""])}>
        {crossOut}
        <span className="PriceComponent__Price">
          {numberWithCommas(price.amount)}
        </span>
        <span className="PriceComponent__Currency">{price.currency}</span>
      </span>
    </span>
  );
};

PriceComponent.displayName = "PriceComponent";
export { PriceComponent };
