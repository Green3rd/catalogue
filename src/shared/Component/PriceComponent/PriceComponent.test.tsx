import { shallow } from 'enzyme';
import { PriceComponent } from './PriceComponent';
import enzymeToJson from 'enzyme-to-json';
import * as PriceHelper from './priceHelper';

const setup = () => {
  const ownProps = {
    price: {
      amount: 55.5,
      currency: '$',
    }
  };

  const props = {
    ...ownProps,
  };

  const Component = shallow(<PriceComponent {...props} />);
  return { Component, props };
};

describe('<PriceComponent />', () => {
  it('should render and function correctly', () => {
    const spyNumberWithCommas = jest.spyOn(PriceHelper, 'numberWithCommas');
    spyNumberWithCommas.mockReturnValue('55.5$');
    const { Component, props } = setup();
    expect(spyNumberWithCommas).toHaveBeenCalledTimes(1);
    expect(spyNumberWithCommas).toHaveBeenCalledWith(props.price.amount);
    expect(enzymeToJson(Component)).toMatchSnapshot();
  });
});
