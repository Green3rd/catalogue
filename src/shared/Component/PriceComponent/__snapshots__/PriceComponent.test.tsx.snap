// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PriceComponent /> should render and function correctly 1`] = `
<span
  className="PriceComponent"
>
  <span
    className="PriceComponent__Price"
  >
    55.5$
  </span>
  <span
    className="PriceComponent__Currency"
  >
    $
  </span>
</span>
`;
